### ISSUE TITLE
mj-section overlapping content in outlook on windows

### ISSUE THREAD

**Issue by luke-nehemedia** [mjmlio/mjml#1986]
I have made a simple mail design consisting of a full-width image on top, implemented as `mj-section` with a background-image. The design works fine on almost all mail clients except outlook on Windows (on Mac it works fine). 

On Outlook the section shows as expected but overlaps the content that would come after it. So it basically "floats" above. When resizing the window Outlook re-renders and it works fine. When switching to another mail and back to the first one, it is once again overlapping the content. 

**What I have tried so far:**
- Setting width and height on `mj-section`
- removing `fullwidth` from the section
- replacing `padding` with an `mj-spacer`
- updating to the latest miml beta

**MJML environment (please complete the following information):**
 - OS: Apache
 - MJML Version: v4.7.0-beta.1
 - MJML tool used: miml compiler 

**Email sending environment(for rendering issues)**:
 - normal sendmail  

**Affected email clients (for rendering issues):**
 - Email Client: Outlook
 - OS: Windows
 - Browser: 

**Screenshots**
[Screenshot (some information blacked out)](https://www.dropbox.com/s/blo8qsysxrdrm49/mjml-outlook-problem1.png?dl=0)
After downloading the image everything stays the same.

**MJML-Code**
```
<mj-section full-width="full-width" width="100%" height="300px" background-url="{{titleImage}}" background-width="100%" background-height="300px" vertical-align="middle" background-size="cover" background-repeat="no-repeat" padding="20px 0px 20px 0px" text-align="center">
	<mj-column width="100%" height="300px" vertical-align="middle">
		{% block logo %}{% endblock %}
		<mj-spacer height="220px" />
		<mj-text padding="15px 0px 15px 0px" color="#ffffff" font-family="Open Sans, Arial, sans-serif" align="center" font-size="45px" line-height="65px" font-weight="900">
			<span style="background-color: {{headingcolor}}; padding-left: 13px; padding-right: 13px;">{% block title %}{% endblock %}</span>
		</mj-text>
	</mj-column>
</mj-section>
```

Any ideas how to solve this issue or what else to try?

**Comment by GarryFlemings** [comment]
@luke-nehemedia : Mine is the earliest response. Later responses might be better.

Here's your code, except:
- I reformatted your source code (personal preference; nothing wrong with yours), 
- I substituted your "{{ }}" fields with constants. A photo from our documentation. A random color. Some nonsense text. And,
- I removed attributes that MJML ignored; more later.
No changes that changed rendering.
https://mjml.io/try-it-live/B1g0XZqlD

Here's your code with some changes:
https://mjml.io/try-it-live/SkQsxWcgD

Notes:

- About ignored attributes: https://mjml.io/documentation/#mj-section and https://mjml.io/documentation/#mj-column include complete lists of the attributes supported by those MJML components  (same role in code as an HTML element). MJML ignores other attributes.
- The line I have as "Lorem ipsum dolor sit amet" doesn't appear in the output because it's not in an MJML component. I used an `mj-text` matching yours.
- You expected your code to give "a full-width image on top, implemented as mj-section with a background-image." The `mj-section` `background-url` attribute implements a background image for the text to appear above. You see that. Below, I showed one quick way to get closer to the goal you sought.
- The image I selected is full-width on mobile; not on desktop. One way to get full-width on desktop is to use a larger image. This rendering used the image's native size.

Welcome to MJML. Good luck! 🍀

**Comment by luke-nehemedia** [comment]
Hello @GarryFlemings 

Thank you for the nice reply. 
I adjusted my code according to your changes. As expected, the problem still persists. 

For a better understanding I made a mook-up in photoshop to demonstrate what I am aiming for: [Mookup of expected result](https://www.dropbox.com/s/msi0pawom0917e3/mjml-outlook-mookup.jpg?dl=0)
I am aware that the title image will be cropped - that's fine. I also know that there are problems with `background-repeat` in Outlook. I choose an image large enough to reduce those problems on most screens and the client is fine with having a repeating image just for the users with Outlook. However, the overlapping problem is obviously something that is more serious. 

Then I just copied your example code into a document and sent it (I just removed one of the text lines in the image and added a little more text below). The problem once again persists in Outlook: [Screenshot of sample code (in Outlook on Windows)](https://www.dropbox.com/s/vhd73utsrwfqa0l/mjml-test-screenshot.png?dl=0). On other devices/clients it looks just fine. 

Unfortunately I do not have a second system to try the code on Outlook. I start wondering whether it is just a problem of that particular system. On the other hand I know that Outlook on windows is quite problematic sometimes.

**Here is my updated code:**
I kept the Twig Code in place and just added some explanation to it. 
```
<mj-body background-color="#F4F4F4" 
  color="#55575d" 
  font-family="Open Sans, Arial, sans-serif" 
  width="900px">
    <mj-section 
      full-width="full-width" 
      background-url="{{title image}}" 
      vertical-align="middle" 
      background-size="cover" 
      background-repeat="no-repeat"
      padding="20px 0px 20px 0px"
      text-align="center">
    	<mj-column width="100%" vertical-align="middle">
    		{% block logo %} basically an <mj-image /> {% endblock %}
    		<mj-spacer height="220px" />
    		<mj-text padding="15px 0px 15px 0px" 
    		  color="#ffffff" 
    		  font-family="Open Sans, Arial, sans-serif" 
    		  align="center" 
    		  font-size="45px" 
    		  line-height="65px" 
    		  font-weight="900">
    			<span style="background-color: {{headingcolor}}; padding-left: 13px; padding-right: 13px;">{% block title %}Just the title of the newsletter{% endblock %}</span>
    		</mj-text>
    	</mj-column>
    </mj-section>
	
	<mj-section>
		<mj-column>
			<mj-spacer height="7px" />
		</mj-column>
	</mj-section>

	
	{% block main %} Here the main content is inserted {% endblock %}

	<mj-section>
		<mj-column>
			<mj-spacer height="7px" />
		</mj-column>
	</mj-section>

      //below this the Footer starts...
```

Thanks a lot for your help!

**Comment by GarryFlemings** [comment]
Good! Thanks for the image of the target.

This picture looks like two sections are appropriate. Yup. Your code above does that. I see where the "logo" and "Newsletter" elements come from. I don't see reason for them to fail. I understand the rest of the code, too.

Outlook: Well, there a number of them. The one on Apple devices works well. Some of the others are based on an Internet Explorer rendering engine. Others are based on an MS Word rendering engine. Many have idiosyncrasies. Email developers seem to do lots of testing around Outlook. Sigh. Welcome to email.

I'm guessing you're using MJML via NodeJS. A good way. Two others of interest, both free.

Visual Studio Code has an extension that renders MJML pretty much as you type. Nice. Perhaps it doesn't integrate with Twig; perhaps it would only be useful if you could add the Twig fields after building the email.

The MJML app runs entirely on your computer. https://mjmlio.github.io/mjml-app/

You've already seen Try-It-Live. Larger list at https://github.com/mjmlio/mjml#applications-and-plugins

🍀

**Comment by iRyusa** [comment]
Can you try to send the compiled version with https://putsmail.com/ ( **uncheck** the minifier) if you have the same result ? I'm not really sure how it can happens as background url should be pretty safe to use with current vml implementation on mj-section. You could give a try to mj-hero too maybe.

What version of Outlook you're using ?

**Comment by iRyusa** [comment]
Can't seem to reproduce on EOA with @GarryFlemings cleaned version https://app.emailonacid.com/app/acidtest/6QmeusHsqBmqM46TKcaZ1zowq02uax2i6PhaIOu34W8bz/list all desktop outlook seems to behave as you would expect from it.

**Comment by GarryFlemings** [comment]
@luke-nehemedia : Maybe you have all you need and I needn't write. Just in case:

I hadn't gotten much from the screenshots before. I just went back and understood the first one better.
mjml-outlook-problem1.png from the first message
I'll compare to the first half of https://mjml.io/try-it-live/SkQsxWcgD, which you'll recall is my version of your code.

As to blacking info out: Always okay. And my high school German was a while back. Today, it only helps me identify the language!

In the .png, the object toward the top with the red X square in it: Any chance that results from Outlook finding no image file? Many email clients (yes, many of the Outlooks) require fully qualified paths to a server on the Internet or to the email recipient's computer (good luck with that!); they don't support relative paths. It looks like it tried to render you a picture and it shows the size it would have filled.

Overlaid on that area is "Newsletter-Anmeldung". I take that as equivalent to my "consectetaur ..." line. MJML was supposed to do that.

Below that area is "Falls du dich ...". That appears in a position after the area reserved for the image, it's outside the `mj-section` with the image. I'll guess that's one of the content blobs coming in via Twig. We didn't see that in the first code.

Bottom line, consistent with iRyusa (which is entirely expected, by the way!): From an MJML perspective, it looks like Outlook is working as designed. 🍀 Let us know about further questions! 😀

**Comment by luke-nehemedia** [comment]
Thank you for all your replies. 

@GarryFlemings: You are right, I use MJML via NodeJS. It is integrated into a website running CraftCMS. 

@iRyusa: I did test it but I cannot see the previews. However, everything what I have seen suggests that everything is fine. Thank you for posting the image of the testing. 

To make sure it is not only a problem with my client, I have now send @GarryFlemings version to a bunch of testers. I hope to have some results tomorrow. I will keep you updated. 

Thank you so far for all your help!

**Comment by iRyusa** [comment]
Mostly come from the way you send the email, try to send it with https://putsmail.com/ as we can't reproduce your issue

**Comment by luke-nehemedia** [comment]
I have done some more testing.

I tested it on multiple machines. All machines except those of my client worked fine with different outlook versions (365 abo, 2006, 2013, 2019). The only version that shows the problem is version 2016 (16.0.5032.1000). This is very strange to me, since as far as I understood, the codebase is the same since 2016 for version 2016, 2019, 365abo. 

I tested both, emails sent via the server and via putsmail - always with the same result. I also tested the "cleaned version" and the "full version" via both platforms. I also checked whether my client could receive e-mails via another domain other than their companies domain, to rule out that the mail server changes anything. On the clients pc, the problem was persistent for `@clientdomain.com`, `@gmx.de`, `@web.de` and `@gmail.com`. 

Right now I have no idea anymore what's the cause of this problem.

**Comment by GarryFlemings** [comment]
@luke-nehemedia : Good information. Maybe a little more?

Is the current problem entirely the same as the image you sent a week ago? 
https://www.dropbox.com/s/vhd73utsrwfqa0l/mjml-test-screenshot.png?dl=0

If not, maybe send more information about the current problem? Brief description might be good. A screenshot is always better (remove anything sensitive first).

If you also notice the problem other than Outlook 2016 (16.0.5032.1000), please mention it, too.

For our future discussion, let's always discuss emails that have gone thru Putsmail (without using the minifier). There are lots more possible problems when emails go out without an ESP.

**Comment by luke-nehemedia** [comment]
Yes, the problem is still exactly the same. 

So far we have done a lot of testing, in all cases with _both_ putsmail and the client's mailserver: The problem is persistent only in Outlook 2016. All other versions tested (all mentioned above, plus now also confirmed with version 2013) do not show the problem but work fine. We tested _both_ the cleaned version (from the screenshot) as also the working version (from the other screenshot with removed stuff). Both behave exactly the same with regard to the problem.

**Comment by GarryFlemings** [comment]
Might you post the versions of the code you've most recently experienced the problem with? That could be a big help. Thanks! 🙏

**Comment by iRyusa** [comment]
Well without a proper way to reproduce it, I'm closing. Feel free to reopen
